Improve Readme (#15)
AkramIzz/paginated_live_list: README.md

### `PageCursor` class:

serve as a pointer to the next page. Every page has its own cursor value. if it was null means we want to point the first page (since it has not pereceding pages). the developer can adjust how he want to implement the cursor. this a user's define value. see the example below.

```dart
class IntPageCursor implements PageCursor {
  final int next;

  IntPageCursor(this.next);
}
```

### `Page` Generic class
contains the list `items` of the page and a `cursor` defined by the user. also it has a boolean indentifier to indentify if this is the last page or not.

## Import the package

```dart
import 'package:paginated_live_list/paginated_live_list.dart';
```

## Implementing a PaginationController

 You need to implement `Page` comparison with `PaginationController.compareTo`, `PageCursor` adjustment with `PaginationController.adjustCursor`, and loading of new pages with `PaginationController.onLoadPage`.

`PageCursor` adjustment is needed when a page changes it's cursor due to an update. The adjustment allows the PaginationController to maintain the current pages by not having to reload all the pages after the page that changed it's cursor.

> ### random_controller.dart

Reviewer: use `custom` or `my` instead of `random`

```dart
class RandomController extends PaginationController<Offer> {
    /// YOUR CODE GOES HERE
}
```

### Defining your own cursors

### random_page_cursor.dart
```dart
class RandomPageCursor implements PageCursor {
    final int randomCursor;

    RandomPageCursor(this.randomCursor);
}
```

> then you can use this in your custom controller or where ever you create your cursors.

Reviewer: The user doesn't actually create 

```suggestion
Then you can use it in your controller to implement `onLoadPage` and `adjustCursor`.
```

```dart

> Page<T> createPageCursor(Page<T> page) {
>     final last = page.items.last;
>     int randomNumber = random.nextInt(100)
>     return Page(
>         page.items,
>         RandomPageCursor(DateTime.now().millisecondsSinceEpoch),
>         page.isLastPage);
>   }

Reviewer: remove `randomNumber`, and make the example more realistic (a simplified implmentation of `onLoadPage`)

```suggestion
Stream<Page<T>> onLoadPage(IndexedPageCursor cursor) {
    return dataSource.list(cursor.page, size: 10).map((items) => Page(
      items,
      IndexedPageCursor(cursor.page + 1),
      items.length < 10,
    ));
  }
```

```

> for more informations check the `offer_repository.dart` file on the example folder.

Reviewer: The repository is an implementation detail. Be more specific about the expected information on `offer_repository.dart` file.
```suggestion
for more information on ..... check the `offer_repository.dart` file on the example folder.
```

### `onLoadPage`

> which is a callback for loading new pages. using the defined cursor.

Reviewer: ```suggestion
A callback for loading and streaming pages. The cursor passed is the cursor of the previous page or null if the requested page is the first page.
```

### offers_controller.dart
```dart
@override
  Stream<Page<Offer>> onLoadPage(PageCursor? cursor) {
    return OffersRepository.instance.list(cursor);
  }

> ```

Reviewer: Code from the example directory isn't what we are aiming to include in the readme. You can provide a simpler, more direct implementation as the objective of examples in the readme is to give an idea of what to implement, how to use the package, and to a lesser extent how to implement it (where it makes sense to show the contract of the methods/classes) without going into the details of a good architecture/abstraction and the exact implementation unlike the example which shows a particular usecase (use with firebase), and/or abstraction (Repository).

For example:

```dart
@override
  Stream<Page<Document>> onLoadPage(DocumentPageCursor? cursor) {
    // including `after: cursor.document` makes it clear that the user needs to use the cursor to load the page
    return dataSource.list(after: cursor?.document, limit: 5).map((docs) => Page(
      docs,
      DocumentPageCursor(docs.last),
      // example for how isLastPage is computed. This also shows that the user is responsible of providing this value
      // and how they might compute it.
      docs.length < 5
    ));
  }
```
